In [1]:
import pandas as pd

In [2]:
contributions = pd.read_csv("http://www.firstpythonnotebook.org/_static/contributions.csv")

In [10]:
contributions.head()

calaccess_committee_id                              committee_name  \
0                 1386560  ADULT USE CAMPAIGN FOR PROPOSITION 64; THE   
1                 1386560  ADULT USE CAMPAIGN FOR PROPOSITION 64; THE   
2                 1386560  ADULT USE CAMPAIGN FOR PROPOSITION 64; THE   
3                 1386560  ADULT USE CAMPAIGN FOR PROPOSITION 64; THE   
4                 1386560  ADULT USE CAMPAIGN FOR PROPOSITION 64; THE   

   calaccess_filing_id date_received contributor_lastname  \
0              2083796    2016-09-18              BERGMAN   
1              2083796    2016-09-18                KAHLE   
2              2083796    2016-07-15             MCDEVITT   
3              2083796    2016-08-10      WARREN-SAMARIPA   
4              2083796    2016-07-26      WARREN-SAMARIPA   

  contributor_firstname contributor_city contributor_state contributor_zip  \
0              GRETCHEN    SPRING VALLEY                CA           91978   
1                 MYRNA        SAN DIEGO                CA           92109   
2                   LEO        ESCONDIDO                CA           92025   
3             STEPHANIE        SAN DIEGO                CA           92116   
4             STEPHANIE        SAN DIEGO                CA           92116   

        contributor_employer contributor_occupation  \
0                 A NEW PATH     EXECUTIVE DIRECTOR   
1   NATIONAL SCHOOL DISTRICT                TEACHER   
2              LIFE IONIZERS    SEO/CONTENT MANAGER   
3  STEPHANIE WARREN SAMARIPA           ENTREPRENEUR   
4  STEPHANIE WARREN SAMARIPA           ENTREPRENEUR   

   contributor_is_self_employed  amount  
0                         False    84.0  
1                         False    35.0  
2                         False   198.0  
3                         False   -50.0  
4                          True    50.0

In [11]:
contributions.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 56379 entries, 0 to 56378
Data columns (total 13 columns):
calaccess_committee_id          56379 non-null int64
committee_name                  56379 non-null object
calaccess_filing_id             56379 non-null int64
date_received                   56379 non-null object
contributor_lastname            56375 non-null object
contributor_firstname           53411 non-null object
contributor_city                56369 non-null object
contributor_state               56363 non-null object
contributor_zip                 56366 non-null object
contributor_employer            48572 non-null object
contributor_occupation          53273 non-null object
contributor_is_self_employed    56379 non-null bool
amount                          56379 non-null float64
dtypes: bool(1), float64(1), int64(2), object(9)
memory usage: 5.2+ MB


68 rows of col=calaccess_committee_id...

In [14]:
contributions.calaccess_committee_id.value_counts().reset_index()

index  calaccess_committee_id
0   1302403                   28617
1   1329332                    6054
2   1380675                    5457
3   1380590                    4105
4   1380746                    2662
5   1382067                    2631
6   1346266                    1014
7   1374469                     866
8   1376791                     692
9   1390529                     690
10  1220380                     656
11  1343793                     420
12  1386909                     318
13  1387279                     275
14  1381808                     250
15  1381423                     204
16  1378875                     184
17  1386627                     146
18  1372900                     128
19  1379198                     111
20  1385139                      94
21  1382568                      88
22  1377991                      81
23  1383946                      69
24  1381382                      48
25  1374885                      45
26  1371855                      40
27  1382912                      38
28  1386477                      35
29  1374153                      31
..      ...                     ...
38  1383858                      14
39  1362973                      12
40  1384591                      10
41  1346267                      10
42  1382525                       9
43  1388865                       8
44  1385745                       8
45  1387856                       8
46  1378703                       8
47  1387575                       7
48  1386494                       7
49  1385506                       7
50  1376040                       7
51  1391170                       6
52  1387789                       6
53  1390962                       4
54  1387641                       3
55  1381488                       3
56  1346973                       2
57  1372760                       2
58  1385538                       2
59  1390602                       2
60  1376077                       2
61  1317554                       2
62  1392066                       1
63  1391327                       1
64  1388518                       1
65  1381029                       1
66  1389668                       1
67  1389930                       1

[68 rows x 2 columns]

76 rows of col=committee_name (which should presumably be 68? #3 + #4 =6054)...

In [16]:
contributions.committee_name.value_counts().reset_index()

index  committee_name
0   CALIFORNIA CORRECTIONAL PEACE OFFICERS ASSOCIA...           28617
1   NEWSOM BALLOT MEASURE COMMITTEE; SAFETY FOR AL...            5457
2   YES ON 62, NO ON 66. REPLACE THE COSTLY, FAILE...            4105
3   COMMITTEE TO DEFEAT PROP 23, SPONSORED BY ENVI...            3027
4   SAVE THE BAG BAN, YES ON 67, SPONSORED BY ENVI...            3027
5   STOP PROP 63, A GRASSROOTS, COMMON-SENSE EFFOR...            2662
6   COALITION FOR CIVIL LIBERTIES - NO ON 63, A PR...            2631
7   YES ON PROPOSITION 51 - CALIFORNIANS FOR QUALI...             866
8   YES ON PROP 61, CALIFORNIANS FOR LOWER DRUG PR...             692
9                            VOLUNTEERS FOR YES ON 62             690
10  CALIFORNIANS TO MEND, NOT END, THE DEATH PENAL...             507
11  CALIFORNIANS FOR JUSTICE AND PUBLIC SAFETY, NO...             507
12  CALIFORNIANS FOR RESPONSIBLE MARIJUANA REFORM,...             420
13                OVERTURN CITIZENS UNITED, YES ON 59             318
14  VETO GUNMAGEDDON, CITIZENS WHO ARE OPPOSED TO ...             275
15  YES ON 64, CALIFORNIANS TO CONTROL, REGULATE A...             250
16  YES ON 59, CALIFORNIA CLEAN MONEY ACTION FUND ...             204
17  NO ON PROP 53 - CALIFORNIANS TO PROTECT LOCAL ...             184
18  COMMUNITY COLLEGE FACILITY COALITION ISSUES CO...             164
19  COMMUNITY COLLEGE FACILITY COALITION ISSUES CO...             164
20  COMMUNITY COLLEGE FACILITY COALITION ISSUES CO...             164
21  COMMUNITY COLLEGE FACILITY COALITION ISSUES CO...             164
22  NO ON 57 - STOP EARLY RELEASE OF VIOLENT CRIMI...             146
23  YES ON 67 - PROTECT THE PLASTIC BAG BAN, SPONS...             128
24  NO ON PROP 61 - CALIFORNIANS AGAINST THE DECEP...             111
25  NO ON PROPOSITION 60, CALIFORNIANS AGAINST WOR...              94
26  PUBLIC AND MENTAL HEALTH ADVOCATES AGAINST 64,...              88
27  YES ON 56 - SAVES LIVES CALIFORNIA, A COALITIO...              81
28  NO ON PROP 66, CALIFORNIANS FOR FAIR JUSTICE, ...              69
29  YES ON 55 - CALIFORNIANS FOR BUDGET STABILITY,...              48
..                                                ...             ...
46                             NEW APPROACH PAC (MPO)               9
47                       MAKE THEM LISTEN - YES ON 59               8
48  PROTECT SMALL BUSINESS AND SMOKE FREE ALTERNAT...               8
49       FUND FOR POLICY REFORM (NONPROFIT 501(C)(4))               8
50  CALIFORNIA CALLS ACTION FUND (NONPROFIT 501(C)...               8
51                      YES ON 53 - STOP BLANK CHECKS               7
52  CIVIC PARTICIPATION ACTION FUND - YES ON 57 (N...               7
53  PROTEUS ACTION LEAGUE NON-PROFIT 501 (C)(4) OR...               7
54  DRUG POLICY ACTION - NON PROFIT 501C4, YES ON ...               7
55  MILLION VOTER PROJECT ACTION FUND - YES ON 55,...               6
56  SAM ACTION, INC., A COMMITTEE AGAINST PROPOSIT...               6
57  CALIFORNIA STATEWIDE LAW ENFORCEMENT ASSOCIATI...               5
58  CALIFORNIA STATEWIDE LAW ENFORCEMENT ASSOCIATI...               5
59                        MOMS STRONG AGAINST PROP 64               4
60  CALIFORNIANS FOR HOSPITAL ACCOUNTABILITY AND Q...               3
61  CONSUMER WATCHDOG CAMPAIGN - YES ON 61, MAJOR ...               3
62  COMMON SENSE KIDS ACTION, INC., YES ON PROPS. ...               2
63  SAN FRANCISCO POLICE OFFICERS ASSOCIATION ISSU...               2
64  CALIFORNIANS FOR SENSIBLE REFORM, SPONSORED BY...               2
65  CALIFORNIA KIDS CAMPAIGN, YES ON PROPS 55 & 56...               2
66  CALIFORNIA GROCERS ASSOCIATION ISSUES COMMITTE...               1
67  YES ON 56 STOP CANCER - PLANNED PARENTHOOD OF ...               1
68  PICO CALIFORNIA - YES ON 47 (NON PROFIT 501 (C...               1
69        NO ON 53, NEIGHBORS DEFENDING LOCAL CONTROL               1
70  FWD.US (NONPROFIT 501(C)(4)) IN SUPPORT OF PRO...               1
71  OPEN PHILANTHROPY ACTION FUND YES ON 57 (NONPR...     

258 rows of col=calaccess_filing_id (which should presumably be 68? Or 76?)...

In [18]:
contributions.calaccess_filing_id.value_counts().reset_index()

index  calaccess_filing_id
0    2013304                28612
1    2083720                 3015
2    1979246                 1322
3    2083707                 1292
4    2082835                  999
5    2012736                  974
6    2064720                  930
7    1997835                  894
8    2130864                  857
9    2064020                  802
10   2061823                  762
11   2082426                  740
12   2098615                  707
13   2098423                  694
14   2097541                  686
15   2097584                  681
16   2082753                  646
17   2063076                  576
18   2129919                  543
19   2113515                  508
20   2098515                  501
21   2038645                  482
22   2064651                  464
23   2033917                  445
24   2054483                  432
25   2038035                  419
26   2064114                  388
27   2097199                  375
28   2119171                  318
29   2036255                  302
..       ...                  ...
228  1962528                    1
229  2098375                    1
230  2083027                    1
231  2046840                    1
232  2063087                    1
233  2083513                    1
234  2098623                    1
235  2063925                    1
236  2130939                    1
237  2128516                    1
238  1997827                    1
239  2129387                    1
240  1979746                    1
241  2038246                    1
242  2083497                    1
243  1980342                    1
244  2098417                    1
245  2096593                    1
246  2118879                    1
247  2097974                    1
248  2035803                    1
249  2131129                    1
250  2131654                    1
251  2038272                    1
252  2083470                    1
253  2083758                    1
254  2082735                    1
255  1982896                    1
256  2133918                    1
257  2012153                    1

[258 rows x 2 columns]

590 rows of col=date_received. Vast majority on 16Dec2015. (#562 is Nov2017 in future. #565 is 1000yrs in past. #566 is 100yrs in future. #583 is Aug2018)

In [31]:
contributions.date_received.value_counts().reset_index()

index  date_received
0    2015-12-16          28618
1    2016-11-03            536
2    2016-10-18            518
3    2016-10-03            484
4    2016-06-03            459
5    2016-06-18            414
6    2015-12-18            402
7    2016-09-18            397
8    2015-12-03            394
9    2016-09-03            365
10   2016-10-07            342
11   2016-09-22            305
12   2016-10-12            251
13   2016-09-23            248
14   2016-09-29            240
15   2016-08-12            229
16   2016-11-04            227
17   2016-09-30            222
18   2016-06-24            219
19   2016-09-15            219
20   2016-09-14            211
21   2016-08-31            207
22   2016-09-16            207
23   2016-10-17            205
24   2016-10-11            202
25   2016-10-13            192
26   2016-10-27            186
27   2016-05-03            184
28   2016-08-25            176
29   2016-10-26            176
..          ...            ...
560  2015-03-25              2
561  2015-09-23              2
562  2017-11-04              1
563  2015-01-15              1
564  2015-11-08              1
565  1016-08-03              1
566  2106-08-10              1
567  2015-02-17              1
568  2016-12-07              1
569  2015-10-17              1
570  2015-11-01              1
571  2015-02-28              1
572  2015-02-26              1
573  2015-10-24              1
574  2015-01-23              1
575  2016-12-12              1
576  2015-04-21              1
577  2017-02-07              1
578  2015-03-17              1
579  2015-12-01              1
580  2016-12-02              1
581  2015-03-31              1
582  2015-01-27              1
583  2018-08-28              1
584  2015-10-23              1
585  2015-04-28              1
586  2015-12-27              1
587  2015-01-30              1
588  2015-11-14              1
589  2015-04-16              1

[590 rows x 2 columns]

In [32]:
contributions.amount.value_counts().reset_index()

index  amount
0        287.08   28612
1        100.00    7604
2         25.00    3749
3         50.00    3184
4       1000.00     974
5        250.00     950
6         15.00     920
7         10.00     900
8        500.00     723
9        150.00     556
10        60.00     528
11         5.00     505
12       200.00     499
13        20.00     413
14      5000.00     360
15        75.00     333
16        45.00     297
17        35.00     233
18       185.00     232
19       120.00     227
20        90.00     198
21      2000.00     194
22     10000.00     182
23      2500.00     181
24       130.00     179
25       105.00     161
26        30.00     151
27     50000.00     144
28     25000.00     142
29       300.00     117
..          ...     ...
435      375.00       1
436     1609.75       1
437   137500.00       1
438   205955.00       1
439    -2450.00       1
440   900000.00       1
441      279.00       1
442    29795.50       1
443     2800.00       1
444    25670.50       1
445    -5000.00       1
446    74950.63       1
447    29035.55       1
448       14.14       1
449        3.34       1
450    28000.00       1
451        7.33       1
452       19.43       1
453        6.00       1
454  9956700.00       1
455      817.00       1
456      227.00       1
457    17500.00       1
458    90000.00       1
459  7050000.00       1
460      118.00       1
461    -2500.00       1
462  1105000.00       1
463       67.00       1
464  9904752.00       1

[465 rows x 2 columns]

In [22]:
contrib_amount = contributions.amount.reset_index()

In [24]:
contrib_amount.describe()

index        amount
count  56379.00000  5.637900e+04
mean   28189.00000  7.254136e+03
std    16275.35975  1.588498e+05
min        0.00000 -1.500000e+05
25%    14094.50000  1.000000e+02
50%    28189.00000  2.870800e+02
75%    42283.50000  2.870800e+02
max    56378.00000  1.250000e+07

In [29]:
contrib_amount.sum()

index     1.589268e+09
amount    4.089809e+08
dtype: float64